lib/puppet/util/package/version/rpm.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+require 'puppet/util/rpm_compare'
+
+module Puppet::Util::Package::Version
+  class Error < StandardError; end
+  class Rpm < Numeric
+    # provides Rpm parsing and comparison
+    extend Puppet::Util::RpmCompare
+    include Puppet::Util::RpmCompare
+    include Comparable
+
+    class ValidationFailure < ArgumentError; end
+
+    def self.parse(ver)
+      version = rpm_parse_evr(ver)
+      new(version[:epoch], version[:version], version[:release], version[:arch]).freeze 
+    end
+
+    def to_s
+      version_found = @epoch + ':' unless @epoch.nil?
+      version_found = version_found ? version_found + @version : @version
+      version_found += '-' + @release unless @release.nil?
+      version_found
+    end
+    alias inspect to_s
+
+    def initialize(epoch, version, release, arch)
+      @epoch   = epoch
+      @version = version
+      @release = release
+      @arch    = arch
+    end
+
+    attr_reader :epoch, :version, :release, :arch
+
+    def eql?(other)
+      other.is_a?(Puppet::Util::Package::Version::Rpm) &&
```

Comment:
`other.is_a?(self.class)` makes it easier to move the class/file when/if needed and easier to read.